packages/fuels-core/src/types/raw_slice.rs

```diff
@@ -0,0 +1,73 @@
+use crate::{Parameterize, Token, Tokenizable};
+use fuels_types::errors::Error;
+use fuels_types::param_types::ParamType;
+use std::fmt::{Debug, Display, Formatter};
+
+// To be used when interacting with contracts which have strings in their ABI.
+// The length of a string is part of its type -- i.e. str[2] is a
+// different type from str[3]. The FuelVM strings only support ascii characters.
+#[derive(Debug, PartialEq, Clone, Eq)]
+pub struct RawSlice {
+    data: Vec<u64>,
```

Comment:
I'm not sure if we should encapsulate the data. We don't need to guard it from anything like we needed to for the sized ascii type.

Maybe doing a simple wrapper around the data as we did for `Bits256` and `B512`. Especially because we only care about the data inside and not the wrapper itself, which is a necessary evil for the `Parameterize` and `Tokenizeable` mechanisms to work.

Otherwise, we'd need to implement a bunch of traits to make the usage of this type painless. Like `PartialEq`, `AsRef`, and others.